ydb/docs/en/core/integrations/hibernate.md

```diff
@@ -0,0 +1,237 @@
+```
+
+Or, if you are using programmatic configuration:
+
+```java
+public static Configuration basedConfiguration() {
+    return new Configuration()
+            .setProperty(AvailableSettings.DRIVER, YdbDriver.class.getName())
+            .setProperty(AvailableSettings.DIALECT, YdbDialect.class.getName());
+}
+```
+
+## Usage {#using}
+
+Use this custom dialect just like any other Hibernate dialect. Map your entity classes to database tables and use Hibernate's session factory to perform database operations.
+
+Table of comparison of Java types descriptions with [{{ ydb-short-name }} types](../yql/reference/types/primitive.md):
+
+| Java type                                                           | {{ ydb-short-name }} type   |
+|---------------------------------------------------------------------|-----------------------------|
+| `bool`, `Boolean`                                                   | `Bool`                      |
+| `String`, enum with `@Enumerated(EnumType.STRING)`                  | `Text` (synonym `Utf8`)     |
+| `java.time.LocalDate`                                               | `Date`                      |
+| `java.math.BigDecimal`, `java.math.BigInteger`                      | `Decimal(22,9)`             |
+| `double`, `Double`                                                  | `Double`                    |
+| `float`, `Float`                                                    | `Float`                     |
+| `int`, `java.lang.Integer`                                          | `Int32`                     |
+| `long`, `java.lang.Long`                                            | `Int64`                     |
+| `short`, `java.lang.Short`                                          | `Int16`                     |
+| `byte`, `java.lang.Byte`, enum with `@Enumerated(EnumType.ORDINAL)` | `Int8`                      |
+| `[]byte`                                                            | `Bytes`  (synonym `String`) |
+| `java.time.LocalDateTime`                                           | `Datetime`                  |
+| `java.time.Instant`                                                 | `Timestamp`                 |
+
+Database schema generation based on hibernate entities is supported.
+
+For example, for the Group class:
+
+```java
+@Getter
+@Setter
+@Entity
+@Table(name = "Groups", indexes = @Index(name = "group_name_index", columnList = "GroupName"))
+@NamedQuery(
+        name = "Group.findGroups",
+        query = "SELECT g FROM Group g " +
+                "JOIN Plan p ON g.id = p.planId.groupId " +
+                "JOIN Lecturer l ON p.planId.lecturerId = l.id " +
+                "WHERE p.planId.courseId = :CourseId and l.id = :LecturerId"
+)
+public class Group {
+
+    @Id
+    @Column(name = "GroupId")
+    private int id;
+  
+    @Column(name = "GroupName")
+    private String name;
+  
+    @OneToMany(mappedBy = "group")
+    private List<Student> students;
```

Comment:
It needs to be clarified why this field is here but not in any of the generated SQL below.